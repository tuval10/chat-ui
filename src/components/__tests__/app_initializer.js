import React from 'react';
import {create} from 'react-test-renderer';
import AppInitializer from '../app_initializer';
import {user, serverConnectionState} from '../../test_helpers/fixtures'
import {noop} from 'lodash'

describe('AppInitializer component', () => {
  describe('renders correctly when the app is uninitialized', () => {
    it('when server connection uninitialized', () => {
      const tree = create(
        <AppInitializer
          initialize={noop}
          user={user}
          serverConnection={serverConnectionState['uninitialized']}
        >
          <div>should not see this div</div>
        </AppInitializer>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('when did not connect to the server yet', () => {
      const tree = create(
        <AppInitializer
          initialize={noop}
          user={user}
          serverConnection={serverConnectionState['connectionInitialized']}
        >
          <div>should not see this div</div>
        </AppInitializer>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('when there is no user', () => {
      const tree = create(
        <AppInitializer
          initialize={noop}
          serverConnection={serverConnectionState['connected']}
        >
        <div>should not see this div</div>
        </AppInitializer>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('renders correctly when the app is initialized - including children', () => {
    const tree = create(
      <AppInitializer
        initialize={noop}
        user={user}
        serverConnection={serverConnectionState['connected']}
      >
        <div>should see this div</div>
      </AppInitializer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});