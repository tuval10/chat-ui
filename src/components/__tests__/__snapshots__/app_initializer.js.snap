// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInitializer component renders correctly when the app is initialized - including children 1`] = `
<div>
  <div>
    should see this div
  </div>
</div>
`;

exports[`AppInitializer component renders correctly when the app is uninitialized when did not connect to the server yet 1`] = `
<div
  className="loading"
>
  loading...
</div>
`;

exports[`AppInitializer component renders correctly when the app is uninitialized when server connection uninitialized 1`] = `
<div
  className="loading"
>
  loading...
</div>
`;

exports[`AppInitializer component renders correctly when the app is uninitialized when there is no user 1`] = `
<div
  className="loading"
>
  loading...
</div>
`;
